python/sglang/srt/managers/schedule_batch.py

```diff
@@ -86,6 +88,183 @@
 
 logger = logging.getLogger(__name__)
 
+MM_FEATURE_CACHE_SIZE = 2 * 1024 * 1024 * 1024
+
+
+SHM_LOCK_FILE = "/tmp/shm_wr_lock.lock"
+
+
+class ShmSyncBuffer:
+    def __init__(self, byte_size: int = 4):
+        self.buffer = shared_memory.SharedMemory(create=True, size=byte_size)
+        self.buffer_wrapper = np.ndarray(1, dtype=np.float32, buffer=self.buffer.buf)
+        self.buffer_wrapper *= 0
+        self.meta_data = {
+            "handle": self.buffer.name,
+            "shape": self.buffer_wrapper.shape,
+            "dtype": str(self.buffer_wrapper.dtype),
+        }
+
+    def __del__(self):
+        if isinstance(self.buffer, shared_memory.SharedMemory):
+            self.buffer.close()
+            self.buffer.unlink()
+
+
+class MmItemMemoryChunk:
```

Comment:
Can we move it into an independent file? Since this part can be decoupled with Scheduler.